### PR TITLE
Product of Array Except self (JAVA)

### PR DESCRIPTION
**Pull request by rakshakannu**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added code for Product of Array Except self in java, and updated the README file.

### Checklist

- [yes] I've read the contribution guidelines.
- [yes] I've checked the issue list before deciding what to submit.
- [yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

PR for #5223

### CODE REVIEW

**Comment by rakshakannu** [comment]
May I know if any changes have to be made for this PR? @ankitaggarwal23 
also, would like to know why it has not been merged. thank you.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

Use proper display message how you are reading and writing outputs

File: Java/other/ArrayProductExceptSelf.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*
+Given an array arr[] of n integers, construct a Product Array prod[] (of same size) such that prod[i] is equal to the product of all the elements of arr[] except arr[i]
+*/
```
Comment (ankitaggarwal23) [inline Java/other/ArrayProductExceptSelf.java:3]: - Don't write in such a way it doesn't look good 
- Add a beginner-friendly description at the top

File: Java/other/ArrayProductExceptSelf.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
```
Comment (ankitaggarwal23) [inline Java/other/ArrayProductExceptSelf.java:4]: header missing

File: Java/other/ArrayProductExceptSelf.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+OUTPUT:
+120 60 40 30 24 
+
+INPUT:
+3
+1 0 4
+
+OUTPUT:
+0 4 0 
+
```
Comment (ankitaggarwal23) [inline Java/other/ArrayProductExceptSelf.java:95]: remove extra test case max 2 is enough 
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF
Use proper display message how you are reading and writing outputs

Add input-output prompts to make the program more interactive.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Last Reminder, Read the instructions mentioned on Gssoc discord yesterday. Do the changes as per that.

**Comment by agamgupta2015** [comment]
please assign this issue to me

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity. Thank you for your valuable time.😊

**Pull request closed without merging**